src/server/plugins/engine/components/helpers/components.ts

```
@@ -191,10 +194,11 @@ export function createComponent(
     case ComponentType.HiddenField:
       component = new Components.HiddenField(def, options)
       break
-  }
 
-  if (typeof component === 'undefined') {
-    throw new Error(`Component type ${def.type} does not exist`)
+    default:
+      // @ts-expect-error - may be a type that isn't in ComponentType
+      logger.error(`Component type ${def.type} does not exist`)
+      component = new Components.TextField(def, options)
```

I'm not sure this is the right approach - it could result in components silently falling back on Text Fields if people aren't proactively monitoring logs.

As far as Designer goes, this could be viable style of approach. But the plugin shouldn't be capturing data if the component isn't valid.